sources/driverdog/src/main.rs

```diff
@@ -292,10 +308,11 @@ fn load_modules_sets(
             .get(&target)
             .context(error::MissingModuleSetSnafu { target })?;
         load_modules(driver_config)?
-    }
-    // Load all the modules sets if no target module was given
-    for driver_config in modules_sets.values() {
-        load_modules(driver_config)?;
+    } else {
+        // Load all the modules sets if no target module was given
+        for driver_config in modules_sets.values() {
+            load_modules(driver_config)?;
+        }
```

Comment:
This looks like a behavior change, since previously it would run unconditionally.

If the behavior needs to change, please split that into a separate commit with an explanation of the fix.